src/renderer/coremods/contextMenu/index.tsx

```diff
@@ -84,57 +84,74 @@ type ContextMenuData = ContextMenuProps["ContextMenu"] & {
-  // Using `await import(...)` is undesirable because the new items will only appear once the menu is interacted with
-  const { MenuGroup } = getByProps<Record<string, React.ComponentType>>([
-    "Menu",
-    "MenuItem",
-    "MenuGroup",
-  ])!;
-
-  if (!MenuGroup) return;
+  // Or MenuGroup Component is not available
+  if (!menuItems[navId] || !MenuGroup) return <ContextMenu {...menu} plugged={true} />;
 
   // The data as passed as Arguments from the calling function, so we just grab what we want from it
   const data = menu.data[0];
 
-  const repluggedGroup = <MenuGroup />;
-  repluggedGroup.props.id = "replugged";
-  repluggedGroup.props.children = [];
+  //Add group only if it doesn't exist
+  if (!menu.children.some((child) => child?.props?.id === "replugged")) {
+    const repluggedGroup = <MenuGroup />;
+    repluggedGroup.props.id = "replugged";
+    repluggedGroup.props.children = [];
+
+    // Add in the new menu items right above the DevMode Copy ID
+    // If the user doesn't have DevMode enabled, the new items will be at the bottom
+    menu.children.splice(-1, 0, repluggedGroup);
+  }
+
+  //get sections from where to clean items
+  const usedSectionIds = menuItems[navId]
+    .map((item) => item.sectionId)
+    .filter((item, index, array) => array.indexOf(item) === index);
 
-  // Add in the new menu items right above the DevMode Copy ID
-  // If the user doesn't have DevMode enabled, the new items will be at the bottom
-  menu.children.splice(-1, 0, repluggedGroup);
+  //cleaning old items before adding new ones
+  usedSectionIds.forEach((sectionId) => {
+    try {
+      if (menu.children.at(sectionId)?.props?.children) {
+        menu.children.at(sectionId)!.props.children = Array.isArray(
+          menu.children.at(sectionId)?.props.children,
+        )
+          ? menu.children
+              .at(sectionId)
+              ?.props.children.filter((child: React.ReactElement) => child.props.plugged)
+          : [menu.children.at(sectionId)?.props.children];
+      }
+    } catch (err) {
+      logger.error("Error while removing old menu items", err, menu.children.at(sectionId));
+    }
+  });
 
+  //adding new items
   menuItems[navId].forEach((item) => {
     try {
-      const res = makeItem(item.getItem(data, menu)) as ContextItem & { props: { id?: string } };
+      const itemRet = makeItem(item.getItem(data, menu)) as ContextItem & {
+        props: { replug?: boolean };
+      };
+
+      // removed the code to add unique id every time because it made experience bad while hovering over item and it getting updated
+      // should be fine because we actively keep removing old items ig
 
-      if (res?.props) {
-        // add in unique ids
-        res.props.id = `${res.props.id || "repluggedItem"}-${Math.random()
-          .toString(36)
-          .substring(2)}`;
+      if (itemRet?.props) {
+        // add in prop  for cleanup
+        itemRet.props.replug = true;
```

Comment:
`replug` prop is not needed

Reply:
It is never checking if this prop exists or not

Reply:
check on the filter above this

Reply:
No. That filter is checking for the `plugged` property, not `replug`.

Reply:
yeah I messed up and fixed now ig

Reply:
This is causing issues with submenus.